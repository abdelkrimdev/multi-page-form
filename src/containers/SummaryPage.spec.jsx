import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import ConnectedSummaryPage, { SummaryPage } from './SummaryPage'

const mockStore = configureStore()

describe('summary page', () => {
  it('should render without crashing.', () => {
    const wrapper = shallow(
      <SummaryPage form={{ step: 0 }} />
    )

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<SummaryPage form={{ step: 0 }} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should connect to the store without crashing.', () => {
    const store = mockStore({
      form: { step: 0 }
    })

    const container = shallow(<ConnectedSummaryPage store={store} />).dive()

    expect(container).toBeDefined()
  })
})
