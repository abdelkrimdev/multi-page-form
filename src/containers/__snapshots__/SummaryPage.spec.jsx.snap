// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`summary page should render correctly. 1`] = `
<Fragment>
  <div
    className="not-registered"
  >
    Please click 
    <Link
      replace={false}
      to="/registration"
    >
      here
    </Link>
     to register yourself.
  </div>
</Fragment>
`;
